src/go/k8s/Makefile

```diff
@@ -172,7 +172,7 @@ KUTTL ?= $(LOCALBIN)/kubectl-kuttl
 GOLANGCI_LINT ?= $(LOCALBIN)/golangci-lint
 
 ## Tool Versions
-KUSTOMIZE_VERSION ?= v4.5.7
+KUSTOMIZE_VERSION ?= v5.3.0
```

Comment:
Can you change 
https://github.com/redpanda-data/redpanda-operator/blob/3310bacc8ad5b8567f7baf1f38d365ecb695e951/taskfiles/dev.yml#L21